### PR TITLE
Real mimic whole data structure from Smart Proxy

### PR DESCRIPTION
**Pull request by joselsegura**
# Description

Some mismatches between aggregator mock response and Smart Proxy ones regarding the upgrade-risks-prediction endpoint were found during the UI development.

This PR includes some of the detected mismatches and some new ones detected when checked the real response compared with real Smart Proxy.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Testing steps

Tested locally

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by tisnik — APPROVED** [review]
ok

File: README.md [hunk 1]
```diff
@@ -773,10 +773,10 @@ Response from the service:
 ```
 < HTTP/1.1 200 OK
 < Content-Type: application/json; charset=utf-8
-< Date: Wed, 15 Feb 2023 08:44:11 GMT
-< Content-Length: 109
+< Date: Thu, 13 Apr 2023 11:36:01 GMT
+< Content-Length: 187
 < 
-{"status":"ok","upgrade_recommendation":{"upgrade_recommended":true,"upgrade_risks_predictors":{"alerts":[],"operator_conditions":[]}}}
+{"meta":{"last_checked_at":"2023-04-13T11:35:41Z"},"status":"ok","upgrade_recommendation":{"upgrade_recommended":false,"upgrade_risks_predictors":{"alerts":[],"operator_conditions":[]}}}
```
Comment (juandspy) [inline README.md:779]: why is this false?

File: server/upgrade_risks_prediction.go [hunk 2]
```diff
@@ -90,12 +93,40 @@ func (server *HTTPServer) upgradeRisksPrediction(writer http.ResponseWriter, req
...
+				prediction.Predictors.OperatorConditions,
+				types.OperatorCondition{
+					Name:      "foc1",
+					Condition: "condition1",
+					Reason:    "Reason1",
+				},
+				types.OperatorCondition{
+					Name:      "foc2",
+					Condition: "condition2",
+					Reason:    "Reason2",
```
Comment (juandspy) [inline server/upgrade_risks_prediction.go:119]: please include all options for the conditions:
```
  'Degraded',
  'Failing',
  'Not Available',
  'Not Upgradeable',
```

File: server/upgrade_risks_prediction.go [hunk 3]
```diff
@@ -90,12 +93,40 @@ func (server *HTTPServer) upgradeRisksPrediction(writer http.ResponseWriter, req
...
+				prediction.Predictors.Alerts,
+				types.Alert{
+					Name:      "alert1",
+					Namespace: "namespace1",
+					Severity:  "info",
+				},
+				types.Alert{
+					Name:      "alert2",
+					Namespace: "namespace2",
+					Severity:  "warning",
```
Comment (juandspy) [inline server/upgrade_risks_prediction.go:106]: please add another one with `critical` severity

**Review by juandspy — CHANGES_REQUESTED** [review]

File: openapi.json [hunk 4]
```diff
@@ -970,20 +970,52 @@
...
+                    },
+                    "meta": {
+                      "type": "object",
+                      "properties": {
+                        "last_checked_at": {
+                          "type": "string",
+                          "format": "date",
+                          "example": "2020-01-23T16:15:59.478901889Z"
+                        }
+                      }
```
Comment (juandspy) [inline openapi.json:1018]: this is also missing the 204, 503 and 404 response codes

File: README.md (hunk 1 above)
Comment (joselsegura) [reply README.md:779]: Unhandled variable after recovering it. Fixing on the next commit

File: server/upgrade_risks_prediction.go (hunk 3 above)
Comment (joselsegura) [reply server/upgrade_risks_prediction.go:106]: Ok

File: server/upgrade_risks_prediction.go (hunk 2 above)
Comment (juandspy) [reply server/upgrade_risks_prediction.go:119]: it's the `condition` field: https://github.com/RedHatInsights/insights-results-smart-proxy/blob/599bb9943786847ec22c8c6dcdc2c9e5c8fc89fb/server/api/v2/openapi.json#L306

File: openapi.json [hunk 5]
```diff
@@ -970,25 +970,64 @@
...
                 }
               }
             }
+          },
+          "204": {
+            "description": "Status NoContent: happens when the cluster is managed. The upgrade risks prediction is not available for this kind of clusters."
+          },
+          "404": {},
+          "503": {
+            "description": "Status NotAvailable: happens when the AMS API or the Upgrade Risks Prediction service are not available."
```
Comment (juandspy) [inline openapi.json:1030]: are -> is :)

**Review by juandspy — APPROVED** [review]
LGTM

**Pull request merged**